tools/uniform_sampling.cpp

```diff
@@ -130,13 +150,29 @@ main (int argc, char** argv)
 
   // Parse the command line arguments for .pcd files
   vector<int> p_file_indices;
-  p_file_indices = parse_file_extension_argument (argc, argv, ".pcd");
+  vector<std::string> extension;
+  extension.push_back (".pcd");
+  extension.push_back (".ply");
+  extension.push_back (".vtk");
+  p_file_indices = parse_file_extension_argument (argc, argv, extension);
+
   if (p_file_indices.size () != 2)
   {
-    print_error ("Need one input PCD file and one output PCD file to continue.\n");
+    print_error ("Need one input file and one output file to continue.\n");
     return (-1);
   }
 
+  // Check if the input point cloud exists
+  std::string input_filename = argv[p_file_indices[0]];
+  ifstream ifile (input_filename.c_str ());
+  if (!ifile) {
+    print_error ("Cannot found input file name (%s).\n", input_filename.c_str ());
```

Comment:
We can move this to line 75, where `pcl::io::load` returns read status. Otherwise we are opening input file two times.